Description: Remove third-party packages from repository
 The minimal chapel package is not intended to include these packages.
 .
 This is a change to comply with Debian packaging standards, and is not
 meant for upstream.
Author: Ben Albrecht <balbrecht@cray.com>
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makefile
+++ b/Makefile
@@ -123,11 +123,8 @@
 	fi
 
 chplvis: compiler third-party-fltk FORCE
-	cd tools/chplvis && $(MAKE)
-	cd tools/chplvis && $(MAKE) install
 
 third-party-fltk: FORCE
-	cd third-party/fltk && $(MAKE)
 
 clean: FORCE
 	cd compiler && $(MAKE) clean
@@ -154,7 +151,6 @@
 	cd modules && $(MAKE) clobber
 	cd runtime && $(MAKE) clobber
 	cd third-party && $(MAKE) clobber
-	cd tools/chplvis && $(MAKE) clobber
 	-@[ -d doc/sphinx ] && cd doc/sphinx && $(MAKE) clobber
 	rm -rf bin
 	rm -rf lib
--- a/make/Makefile.base
+++ b/make/Makefile.base
@@ -148,17 +148,6 @@
 #
 THIRD_PARTY_DIR = $(CHPL_MAKE_HOME)/third-party
 
-include $(THIRD_PARTY_DIR)/gasnet/Makefile.include
-include $(THIRD_PARTY_DIR)/massivethreads/Makefile.include
-include $(THIRD_PARTY_DIR)/qthread/Makefile.include
-include $(THIRD_PARTY_DIR)/jemalloc/Makefile.include
-include $(THIRD_PARTY_DIR)/gmp/Makefile.include
-include $(THIRD_PARTY_DIR)/hwloc/Makefile.include
-include $(THIRD_PARTY_DIR)/re2/Makefile.include
-include $(THIRD_PARTY_DIR)/llvm/Makefile.include-$(CHPL_MAKE_LLVM)
-include $(THIRD_PARTY_DIR)/fltk/Makefile.include
-include $(THIRD_PARTY_DIR)/libunwind/Makefile.include
-
 -include $(THIRD_PARTY_DIR)/Makefile.devel.include
 
 #
--- a/third-party/Makefile
+++ b/third-party/Makefile
@@ -2,7 +2,6 @@
 export CHPL_MAKE_HOME=$(shell pwd)/..
 endif
 include $(CHPL_MAKE_HOME)/make/Makefile.base
-include $(THIRD_PARTY_DIR)/chpl-venv/Makefile.include
 
 ifdef CHPL_DEVELOPER
 DEBUG=1
@@ -16,61 +15,25 @@
 
 default: all
 
-all: test-venv chpldoc-venv gasnet gmp hwloc jemalloc libunwind qthread llvm
+all:
 
 clean: FORCE
-	cd chpl-venv && $(MAKE) clean
-	cd gasnet && $(MAKE) clean
-	cd gmp && $(MAKE) clean
-	cd hwloc && $(MAKE) clean
-	cd jemalloc && $(MAKE) clean
-	cd libunwind && $(MAKE) clean
-	cd llvm && $(MAKE) clean
-	cd massivethreads && $(MAKE) clean
-	cd qthread && $(MAKE) clean
-	cd re2 && $(MAKE) clean
-	cd fltk && $(MAKE) clean
 
 cleanall: FORCE
-	cd chpl-venv && $(MAKE) cleanall
-	cd gasnet && $(MAKE) cleanall
-	cd gmp && $(MAKE) cleanall
-	cd hwloc && $(MAKE) cleanall
-	cd jemalloc && $(MAKE) cleanall
-	cd libunwind && $(MAKE) cleanall
-	cd llvm && $(MAKE) cleanall
-	cd massivethreads && $(MAKE) cleanall
-	cd qthread && $(MAKE) cleanall
-	cd re2 && $(MAKE) cleanall
-	cd fltk && $(MAKE) cleanall
 
 clobber: FORCE
-	cd chpl-venv && $(MAKE) clobber
-	cd gasnet && $(MAKE) clobber
-	cd gmp && $(MAKE) clobber
-	cd hwloc && $(MAKE) clobber
-	cd jemalloc && $(MAKE) clobber
-	cd libunwind && $(MAKE) clobber
-	cd llvm && $(MAKE) clobber
-	cd massivethreads && $(MAKE) clobber
-	cd qthread && $(MAKE) clobber
-	cd re2 && $(MAKE) clobber
-	cd fltk && $(MAKE) clobber
 
 depend:
 
 test-venv: $(CHPL_VENV_TEST_REQS)
 $(CHPL_VENV_TEST_REQS): $(CHPL_VENV_TEST_REQUIREMENTS_FILE)
-	cd chpl-venv && $(MAKE) test-venv
 
 chpldoc-venv: $(CHPL_VENV_SPHINX_BUILD)
 $(CHPL_VENV_SPHINX_BUILD): $(CHPL_VENV_CHPLDOC_REQUIREMENTS_FILE)
-	cd chpl-venv && $(MAKE) chpldoc-venv
 
 # See gasnet/Makefile for explanation of the post-install step
 gasnet: $(GASNET_INSTALL_DIR)
 $(GASNET_INSTALL_DIR):
-	cd gasnet && $(MAKE) && $(MAKE) post-install
 
 try-gmp: FORCE
 ifeq ($(wildcard $(GMP_BUILD_DIR)),)
@@ -82,23 +45,18 @@
 
 gmp: $(GMP_H_FILE)
 $(GMP_H_FILE):
-	cd gmp && $(MAKE)
 
 hwloc: $(HWLOC_INSTALL_DIR)
 $(HWLOC_INSTALL_DIR):
-	cd hwloc && $(MAKE)
 
 jemalloc: $(JEMALLOC_INSTALL_DIR)
 $(JEMALLOC_INSTALL_DIR):
-	cd jemalloc && $(MAKE)
 	
 libunwind: $(LIBUNWIND_INSTALL_DIR)
 $(LIBUNWIND_INSTALL_DIR):
-	cd libunwind && $(MAKE)
 
 massivethreads: $(MASSIVETHREADS_INSTALL_DIR)
 $(MASSIVETHREADS_INSTALL_DIR):
-	cd massivethreads && $(MAKE)
 
 ifeq ($(CHPL_MAKE_HWLOC), hwloc)
 qthread: $(HWLOC_INSTALL_DIR) $(QTHREAD_INSTALL_DIR)
@@ -106,14 +64,11 @@
 qthread: $(QTHREAD_INSTALL_DIR)
 endif
 $(QTHREAD_INSTALL_DIR):
-	cd qthread && $(MAKE)
 $(QTHREAD_ALIASES): qthread
 
 llvm: FORCE
-	cd llvm && $(MAKE)
 
 libhdfs3: FORCE
-	cd libhdfs3 && $(MAKE)
 
 try-re2: FORCE
 ifeq ($(wildcard $(RE2_BUILD_DIR)),)
@@ -127,7 +82,6 @@
 
 re2: $(RE2_H_FILE)
 $(RE2_H_FILE):
-	cd re2 && $(MAKE)
 
 -include Makefile.devel
 
